code/modules/mob/hearing/virtualhearer.dm

```diff
@@ -79,6 +79,10 @@ var/list/stationary_hearers = list(	/obj/item/device/radio/intercom,
 		sight |= adding
 	if(removing)
 		sight &= ~removing
+	if(sight & (SEE_TURFS|SEE_MOBS|SEE_OBJS))
```

Comment:
this was the shitcode, too. It was a little too hacked together for my tastes, even for our codebase.